pandas/core/apply.py

```diff
@@ -723,13 +723,18 @@ def apply(self) -> DataFrame | Series:
 
         return self.apply_standard()
 
+    def _filter_numeric_only(self) -> None:
+        if "numeric_only" in self.kwargs and self.kwargs["numeric_only"] is True:
+            self.obj = self.obj._get_numeric_data()
+
     def agg(self):
         obj = self.obj
         axis = self.axis
 
         # TODO: Avoid having to change state
         self.obj = self.obj if self.axis == 0 else self.obj.T
         self.axis = 0
+        self._filter_numeric_only()
```

Comment:
This applies the numeric_only in all cases rather that just where it's necessary. Can you move this to agg_list_like and agg_dict_like.

Reply:
Good point. I will move it.